### PR TITLE
792 : Expire all team assignments when plan status changes to Retired or Completed

### PR DESCRIPTION
**Pull request by rehammuzzamil**
https://github.com/OpenSRP/opensrp-server-web/issues/792

### CODE REVIEW

**Review by dubdabasoduba — DISMISSED** [review]

File: src/main/java/org/opensrp/service/OrganizationService.java [hunk 1]
```diff
@@ -260,4 +260,15 @@ public Organization findOrganizationByName(String organizationName) {
 		return organizationRepository.findOrganizationByName(organizationName);
 	}
 
+	public void unassignLocationAndPlan(String planIdentifier) {
```
Comment (Rkareko) [inline src/main/java/org/opensrp/service/OrganizationService.java:263]: Can we add some javadocs here

File: src/main/java/org/opensrp/repository/postgres/OrganizationRepositoryImpl.java [hunk 2]
```diff
@@ -353,5 +361,17 @@ public Organization findOrganizationByName(String name) {
 		return convert(organizations);
 	}
 
+	@Override
+	public void unassignLocationAndPlan(Long planId) {
```
Comment (Rkareko) [inline src/main/java/org/opensrp/repository/postgres/OrganizationRepositoryImpl.java:367]: Can we add some javadocs here

File: src/main/java/org/opensrp/repository/postgres/OrganizationRepositoryImpl.java [hunk 3]
```diff
@@ -152,6 +152,14 @@ public void assignLocationAndPlan(Long organizationId, String jurisdictionIdenti
 		return organizationLocationMapper.selectByExampleAndDateTo(example.getOredCriteria(), example.getOrderByClause(),
 		    currentDate);
 	}
+
+	private List<OrganizationLocation> getAssignedLocationsByPlanId(Long planId) {
```
Comment (Rkareko) [inline src/main/java/org/opensrp/repository/postgres/OrganizationRepositoryImpl.java:156]: Can we add some javadocs here

Comment (Rkareko) [reply src/main/java/org/opensrp/repository/postgres/OrganizationRepositoryImpl.java:156]: LGTM

File: src/main/java/org/opensrp/repository/postgres/OrganizationRepositoryImpl.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/repository/postgres/OrganizationRepositoryImpl.java:367]: Done @Rkareko

File: src/main/java/org/opensrp/service/OrganizationService.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/service/OrganizationService.java:268]: Done @Rkareko

File: src/main/java/org/opensrp/repository/postgres/OrganizationRepositoryImpl.java (hunk 3 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/repository/postgres/OrganizationRepositoryImpl.java:161]: Added.

**Review by Rkareko — APPROVED** [review]

**Pull request merged**